src/Stratis.Bitcoin.Features.PoA/FederationManager.cs

```diff
@@ -25,9 +26,9 @@ public class FederationManager
 
         public FederationManager(NodeSettings nodeSettings, Network network, PoAMiner miner, ILoggerFactory loggerFactory)
         {
-            this.settings = nodeSettings;
-            this.network = network as PoANetwork;
-            this.miner = miner;
+            this.settings = Guard.NotNull(nodeSettings, nameof(nodeSettings));
```

Comment:
I don't like the name `FederationManager` I dont think we have similar equivalents with POW/POS (like a PowManager)
So you could consider just calling this PoaMiner or something similar (future PR is ok).